desidlas/training/training.py

```diff
@@ -0,0 +1,397 @@
+                             result_loss_offset_regression, result_rmse_coldensity,
+                             result_loss_coldensity_regression))
+                    
+                # Run on test
+                if i % 5000 == 0 or i == training_iters - 1:
+                    test_accuracy, _, result_rmse_offset, _, result_rmse_coldensity, _ = train_ann_test_batch(
+                        sess, np.arange(test_dataset.fluxes.shape[0]), test_dataset.data)
+                    best_accuracy = test_accuracy if test_accuracy > best_accuracy else best_accuracy
+                    best_offset_rmse = result_rmse_offset if result_rmse_offset < best_offset_rmse else best_offset_rmse
+                    best_density_rmse = result_rmse_coldensity if result_rmse_coldensity < best_density_rmse else best_density_rmse
+                    
+                    print("             test accuracy/offset RMSE/density RMSE:     {:0.3f} / {:0.3f} / {:0.3f}\n".format(
+                          test_accuracy, result_rmse_offset, result_rmse_coldensity))
+                    record_file.write("             test accuracy/offset RMSE/density RMSE:     {:0.3f} / {:0.3f} / {:0.3f}".format(
+                          test_accuracy, result_rmse_offset, result_rmse_coldensity))
+                    
+                    save_checkpoint(sess, (save_filename + "_" + str(i)) if save_filename is not None else None)
+
+           # Save checkpoint
+            save_checkpoint(sess, save_filename)
+
+            return best_accuracy, test_accuracy, np.mean(loss_value), best_offset_rmse, result_rmse_offset, \
+                   best_density_rmse, result_rmse_coldensity
+
+def save_checkpoint(sess, save_filename):
+    """
+    save model file
+
+    Parameters
+    ----------
+    sess:tf.Session object
+    save_filename:the path to save the model
+
+    Returns
+    -------
+
+    """
+    if save_filename is not None:
+        tf.compat.v1.train.Saver().save(sess, save_filename + ".ckpt")
+        with open(checkpoint_filename + "_hyperparams.json", 'w') as fp:
+            json.dump(hyperparameters, fp)
+        print("Model saved in file: %s" % save_filename + ".ckpt")
+
+def calc_normalized_score(best_accuracy, best_offset_rmse, best_coldensity_rmse):
+    """
+    Generate the normalized score from the 3 loss functions
+
+    This should be used for a hyperparameter search
+
+    Parameters
+    ----------
+    best_accuracy:the highest classification accuracy during the training
+    best_offset_rmse:the minimum offset RMSE during the training
+    best_coldensity_rmse:the minimum column density during the training
+
+    Returns:
+    -------
+    normalized_score:used to select hyperparameters. A better hyperparameter combine will have a smaller normalized_score
+
+    """
+    # These mean & std dev are used to normalize the score from all 3 loss functions for hyperparam optimize
+    # TODO -- Reconsider these for DESI
+    mean_best_accuracy = 0.02
```

Comment:
Were these optimized?  If not, should they be?

Are these reported in the paper?

Reply:
They are not optimized, just as same as what Parks chose. We do not describe these values in detail in the paper, we just report we do the hyperparameter search.